src/main/java/com/github/alexcojocaru/mojo/elasticsearch/v2/ForkedElasticsearchProcessDestroyer.java

```diff
@@ -113,29 +113,45 @@ private void terminateWindowsProcess(InstanceConfiguration config)
                 config.getId()));
         
         CommandLine command = ProcessUtil.buildKillCommandLine(pid);
-        try
-        {
-            ProcessUtil.executeScript(config, command, true);
-
-            log.info(String.format(
-                    "... the Elasticsearch process [%d] has stopped.",
-                    config.getId()));
-        }
-        catch (Exception e)
-        {
-            if (e.getMessage().contains("no running instance"))
+        for (int retry = 0; retry < 3; ++retry) {
+            try
             {
+                ProcessUtil.executeScript(config, command, true);
+    
                 log.info(String.format(
                         "... the Elasticsearch process [%d] has stopped.",
                         config.getId()));
+                ProcessUtil.cleanupPid(config.getBaseDir());
             }
-            else
+            catch (Exception e)
             {
-                log.error(String.format(
-                        "Error while destroying the Elasticsearch process [%d]",
-                        config.getId()),
-                        e);
+                if (e.getMessage().contains("no running instance"))
+                {
+                    log.info(String.format(
+                            "... the Elasticsearch process [%d] has stopped.",
+                            config.getId()));
+                    ProcessUtil.cleanupPid(config.getBaseDir());
+                }
+                else
+                {
+                    // sometimes taskkill fails with "The operation attempted is not supported."
+                    // https://blogs.technet.microsoft.com/markrussinovich/2005/08/17/unkillable-processes/
+                    // in that case, retry
+                    if (retry < 2) {
+                        try {
+                            Thread.sleep(500);
+                            continue;
```

Comment:
On 1st and 2nd attempt, if the sleep fails, the kill will not be retried (the `continue` will be skipped, and the catch block will be executed instead). Is this what you wanted?

Reply:
The only reason for `Thread.sleep()` to fail is when `interrupt()` is called by another thread. In that case we want to exit asap. So yes it's what I wanted.

Reply:
:+1: